jupyterhub/auth.py

```diff
@@ -37,6 +37,25 @@ class Authenticator(LoggingConfigurable):
     """Base class for implementing an authentication provider for JupyterHub"""
 
     db = Any()
+    
+    enable_auth_state = Bool(False, config=True,
+        help="""Enable persisting auth_state (if available).
+
+        auth_state will be encrypted and stored in the Hub's database.
+        This can include things like authentication tokens, etc.
+        to be passed to Spawners as environment variables.
+
+        Encrypting auth_state requires the cryptography package.
+
+        Additionally, the JUPYTERHUB_CRYPTO_KEY envirionment variable must
+        contain one (or more, separated by ;) 32B encryption keys.
+        These can be either base64 or hex-encoded.
```

Comment:
After thinking about it more, I think *you* are right :)

I have a tendency to try to allow every possible input, which tends to cause headaches down the road. After thinking about this more, I think you're right and we should only allow hex keys because our users aren't going to know or care what a Fernet key is and we already generate 32B hex secrets elsewhere in the docs, so there isn't a *good* reason for this to be different. Consistency with ourselves is probably better than consistency with the docs of an underlying implementation that our users won't read.